### PR TITLE
Add a workflow for indexing

### PR DESCRIPTION
**Pull request by dshevtsov**
## Description

Adding a workflow for manual running of the indexing separately.
As a template for the workflow I use the github-pages workflow.
Not sure about the environmental variables that must be set for the workflow.
I added the list variables from other workflows and removed the variables that I couldn't find in the code of this repo, the aio-theme, and Gatsby docs.

## Motivation and Context

A separate workflow for indexing can allow to trigger indexing form multiple repositories using external automation, e.g. Jenkins.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by jeff-matthews — COMMENTED** [review]
I'll defer to @dmitrymatio on env vars, but I think you're missing the following:

- `GATSBY_ALGOLIA_SEARCH_INDEX=["Adobe Commerce", "All Products"]`
- `REPO_NAME=`

but I was wondering how you think we should trigger the workflow with automation in the future. It looks like you can use a [`schedule`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) event in the workflow similar to how we used to schedule indexing for devdocs.magento.com on Jenkins.

**Comment by dshevtsov** [comment]
> I'll defer to @dmitrymatio on env vars, but I think you're missing the following:
> 
> * (commerce-specific) - `GATSBY_ALGOLIA_SEARCH_INDEX=["Adobe Commerce", "All Products"]`
> * `REPO_NAME=`


`GATSBY_ALGOLIA_SEARCH_INDEX` is missing in the code. It was used before but I didn't find it nor in this repo neither in the theme. i didn't use it as deprecated.
`REPO_NAME` is there.

**Comment by dshevtsov** [comment]
> I was also wondering how you think we should trigger the workflow with automation in the future. It looks like you can use a [`schedule`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) event in the workflow similar to how we used to schedule indexing for devdocs.magento.com on Jenkins via a cron job.

Yes, I was thinking about the cron available in GH workflow too. Also, I want to explore an opportunity to trigger the workflow from Jenkins for cases when unscheduled re-indexing might be needed. And I'm starting the work from a manual run. When I'm sure the indexing works, I'll investigate the cron and Jenkins options.

File: .github/workflows/index.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        env:
+          PREFIX_PATHS: true # equivalent to --prefix-paths flag for 'gatsby build'
+          REPO_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO_OWNER: ${{ github.repository_owner }}
+          REPO_NAME: ${{ github.event.repository.name }}
+          REPO_BRANCH: ${{ github.ref_name }}
+          GATSBY_ALGOLIA_APPLICATION_ID: ${{ secrets.AIO_ALGOLIA_APPLICATION_ID }}
+          GATSBY_ALGOLIA_SEARCH_API_KEY: ${{ secrets.AIO_ALGOLIA_SEARCH_API_KEY }}
+          ALGOLIA_WRITE_API_KEY: ${{ secrets.AIO_ALGOLIA_WRITE_API_KEY }}
+          ALGOLIA_INDEXATION_MODE: index
```
Comment (jeff-matthews) [inline .github/workflows/index.yml:24]: It would be good to use `console` for testing to avoid over using the API.

Comment (dshevtsov) [reply .github/workflows/index.yml:24]: good point

Comment (dmitrymatio) [reply .github/workflows/index.yml]: `GATSBY_ALGOLIA_SEARCH_INDEX` is no longer needed. We've also moved to an external product-index map which you can find along with latest indexing instructions here https://github.com/AdobeDocs/search-indices

**Review by jeff-matthews — APPROVED** [review]

File: .github/workflows/index.yml (hunk 1 above)
Comment (dshevtsov) [reply .github/workflows/index.yml]: @dmitrymatio, is https://github.com/AdobeDocs/dev-site/blob/main/static/shared/search/index.js still relevant? Or https://github.com/AdobeDocs/search-indices replaces it?

**Pull request merged**